llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -2292,8 +2298,10 @@ X86TargetLowering::X86TargetLowering(const X86TargetMachine &TM,
     setOperationAction(ISD::FP_EXTEND,            MVT::f32, Legal);
     setOperationAction(ISD::STRICT_FP_EXTEND,     MVT::f32, Legal);
 
-    setCondCodeAction(ISD::SETOEQ, MVT::f16, Expand);
-    setCondCodeAction(ISD::SETUNE, MVT::f16, Expand);
+    setCondCodeAction(ISD::SETOEQ, MVT::f16,
+                      Subtarget.hasAVX10_2() ? Custom : Expand);
+    setCondCodeAction(ISD::SETUNE, MVT::f16,
+                      Subtarget.hasAVX10_2() ? Custom : Expand);
```

Comment:
Why don't combine them to the loop? You can move them to line 2445.

Reply:
I can do that. Although `setGroup(MVT::f16);` on 2286 line above has better proximity fore related code.

Reply:
The code is in the scope of `hasFP16`. We prefer to organizing them in feature bulk rather than mixing them together.

Reply:
This code movement result in 3 regressions.

Reply:
I don't see how it results in regressions. Did you remove these 2 Expand?

Reply:
Here is link on infra.
https://buildkite.com/llvm-project/github-pull-requests/builds/113138#0192c4a7-5933-4d41-b189-f83fdfe10fd9

Reply:
I meant this:
```
diff --git a/llvm/lib/Target/X86/X86ISelLowering.cpp b/llvm/lib/Target/X86/X86ISelLowering.cpp
index 676ae6b87610..1ffec84d9958 100644
--- a/llvm/lib/Target/X86/X86ISelLowering.cpp
+++ b/llvm/lib/Target/X86/X86ISelLowering.cpp
@@ -2441,6 +2441,11 @@ X86TargetLowering::X86TargetLowering(const X86TargetMachine &TM,
         setOperationAction(ISD::SETCC, VT, Custom);
       }
     }
+
+    for (auto VT : {MVT::f16, MVT::f32, MVT::f64}) {
+      setCondCodeAction(ISD::SETOEQ, VT, Custom);
+      setCondCodeAction(ISD::SETUNE, VT, Custom);
+    }
   }

   if (!Subtarget.useSoftFloat() && Subtarget.hasVLX()) {
```

Notice it's under these code, so it should not affect any existing tests.

Reply:
https://github.com/llvm/llvm-project/pull/113695
https://github.com/llvm/llvm-project/pull/113905,  2 commits on this very branch
I tried out  4 different ways, except current change i get regression for all moves.

Reply:
See https://github.com/llvm/llvm-project/commit/2c9b981b9d8541d7c9e026baff1325e1708caeae
I don't see any problem on my side.
Maybe you was misled. The diff above was to demonstrate the change in function `X86TargetLowering`, so I omitted the change in `LowerSETCC`. I'm surprised you didn't explore along this way.